### ISSUE TITLE
Shared database groups list get synced only after application reboot

### ISSUE THREAD

**Issue by TsypliakAlexandr** [JabRef/jabref#11232]
### JabRef version

Latest development branch build (please note build date below)

### Operating system

Windows

### Details on version and operating system

Windows 10 Pro 22H2 19045.4291

### Checked with the latest development build (copy version output from About dialog)

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

Version used:
JafRef-5.14.msi version from 04/19/2024

Precondition:
Two Windows users got connected to same remote Postgres database

Steps to reproduce:
1. User A creates adds new Groups name "Test"
2. User A sees new group "Test"
3. User B goes to File -> Shared Database and clicks "Pull changes from shared database"

Actual behavior:
1. User B can't see new group "Test" created by user A

Expected behavior:
1. User B sees new group "Test" created by user A

Workaround: 
1. User B closes application GUI, opens it again and connects to same remote Postgres database - then he can see new group "Test" created by user A

**Comment by Siedlerchr** [comment]
Groups are part of the metadata and are synced as well when a new group is added,  at least what I observed. I need to do some more testing to find out where exactly on the receiver side the sync does trigger a change/does not

**Comment by Siedlerchr** [comment]
I debugged and analyzed a bit further, and I think I narrowed down the problem. It's mainly a UI issue.

As a workaround you can create a new empty library and just switch back and forth
1. User A adds a new group to the shared database
2. User B does not see the changes -> Switches to a new empty library and back 
3. Group is there

This works for adding or changing existing groups.

Code related: 
https://github.com/JabRef/jabref/blob/dfe1b0ce0bbd185ed5747b70e7ca24e4f05ebf37/src/main/java/org/jabref/gui/groups/GroupTreeViewModel.java#L143-L158

Groups deletion seems to be a separate issue. That does not get synced correctly in the metadata. At least from my debugging observation
